packages/firebase_storage/ios/Classes/FirebaseStoragePlugin.m

```diff
@@ -88,4 +90,17 @@ - (void)getData:(FlutterMethodCall *)call result:(FlutterResult)result {
             }];
 }
 
+- (void)deleteData:(FlutterMethodCall *)call result:(FlutterResult)result {
+  NSString *path = call.arguments[@"path"];
+  FIRStorageReference *ref = [[FIRStorage storage].reference child:path];
+  [ref deleteWithCompletion:^(NSError *error) {
+    if (error != nil) {
+      result(error.flutterError);
+      return;
+    } else {
+      result(@(YES));
```

Comment:
This can be `nil`, since this is just confirming that the call to the native side was successful. You can generate the `DeleteTaskResult` object based on success or failure of the invocation no need to pass it in here.